R/utils.r

```diff
@@ -59,23 +67,67 @@ any.latex.specials <- function(char_vec) {
 		return(any(grepl(special, char_vec, fixed = TRUE)))
 	}
 	# Search, using vapply, for each special.
-	specials_match <- vapply(names(latex.specials), grepl.one.special,
+	specials_match <- vapply(names(specials_list), grepl.one.special,
 		char_vec = char_vec, FUN.VALUE = FALSE)
 	return(any(specials_match))
 }
 
-latex.escape.vec <- function(vec) {
-	# latex.escape.vec should never change the class of its input.
+.escape.vec <- function(vec, escape_type) {
+	# .escape.vec should never change the class of its input.
 	# That seems useful, since functions like ellip.limit.arr check class.
-	if (!(is.vector(vec) || is.factor(vec))) {
-	    stop('expected `vec` to be a vector or factor but it is a ', paste(class(vec), collapse = ', '))
-	}
-	if (any.latex.specials(vec)) {
+	# escape_type must be 'latex' or 'html'
+	stopifnot(any(escape_type == c('html', 'latex')))
+	# if (!(is.vector(vec) || is.factor(vec))) {
+	#     stop('expected `vec` to be a vector or factor but it is a ', paste(class(vec), collapse = ', '))
```

Comment:
If left in: the comments should get a comment, so that my future forgetful self does not need to ask this question again :-)